Gems/Meshlets/ASV_GPU_and_CPU_Demo/SampleComponentManager.cpp

```diff
@@ -0,0 +1,1672 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <SampleComponentManager.h>
+#include <SampleComponentConfig.h>
+
+#include <Atom/Component/DebugCamera/CameraComponent.h>
+#include <Atom/Component/DebugCamera/NoClipControllerComponent.h>
+#include <Atom/Component/DebugCamera/ArcBallControllerComponent.h>
+#include <Atom/Feature/AuxGeom/AuxGeomFeatureProcessor.h>
+#include <Atom/Feature/ImGui/ImGuiUtils.h>
+#include <Atom/Feature/ImGui/SystemBus.h>
+#include <Atom/Feature/Mesh/MeshFeatureProcessor.h>
+#include <Atom/Feature/PostProcessing/PostProcessingConstants.h>
+#include <Atom/Feature/SkinnedMesh/SkinnedMeshInputBuffers.h>
+
+#include <Atom/RPI.Public/Pass/Pass.h>
```

Comment:
I don't think you meant to add this file, since the Readme instructions already include 'Alter SampleComponentManager.cpp' instead of 'copy this file'.

Reply:
yes to the demo folder - the reason is that by now it has 3 different changes in it and so, while I explain it in the read me files and also have a ready to go branch (Atom/barlev/Meshlets_FinalWorkingVersion), this file can be used as reference to the required changes if required.

Reply:
Remark: I was trying to be over protective / conservative when creating the demo folder as reference to changed and required files. 
By now most of the connectivity is done automatically (mainly the passes injection) or using a side branch in ASV (to have the sample connected).
This can be removed at a later date if decided, but for now it is a good reference to the entire process, and the read me covers how to do it from scratch if needed.